src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java

```diff
@@ -133,7 +133,7 @@ public Bytes<U> clear()
             throws IllegalStateException {
         final long start = start();
         final long capacity = capacity();
-        if (readPosition == start && writePosition == start && writeLimit == capacity)
+        if (readPosition == start && writePosition() == start && writeLimit == capacity)
```

Comment:
Good find, I wonder if it's feasible to/worth making `writePosition` private to reduce the number of places a bug like this can be reintroduced going forward?

Reply:
Good point.
It's a compatibility break to make this change right away - I've left a comment to do it in x.24